Data/HashMap/Base.hs

```diff
@@ -751,12 +953,58 @@ update f = alter (>>= f)
 -- absence thereof. @alter@ can be used to insert, delete, or update a value in a
 -- map. In short : @'lookup' k ('alter' f k m) = f ('lookup' k m)@.
 alter :: (Eq k, Hashable k) => (Maybe v -> Maybe v) -> k -> HashMap k v -> HashMap k v
+-- TODO(m-renaud): Consider using specialized insert and delete for alter.
```

Comment:
This implementation of `alter` is ridiculous indeed. We will need some benchmarks with `Int` keys to make sure we don't get significant regressions by doing the sensible thing here.

Reply:
I'll fix alter as a follow-up to this PR.